wsclient/src/test/java/org/hpccsystems/ws/client/WSECLTests.java

```diff
@@ -0,0 +1,181 @@
+{
+    private static HPCCWsWorkUnitsClient wswuclient = null;
+    private static boolean hasPublishedQuery = false;
+    private final static String wsECLPort = System.getProperty("wseclport", "8002");
+    private final static String eclScriptName = "SimpleFunction.ecl";
+    private static Connection wseclConn = null;
+
+    @BeforeClass
+    public static void setup() throws Exception
+    {
+        wswuclient = wsclient.getWsWorkunitsClient(); //for publishing queries
+        Assert.assertNotNull(wswuclient);
+        
+        try
+        {
+             InputStream resourceStream = BaseRemoteTest.class.getClassLoader().getResourceAsStream(eclScriptName);
+             ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+
+             byte[] buffer = new byte[4096];
+             int bytesRead = resourceStream.read(buffer);
+             while (bytesRead > -1) {
+                 byteArrayOutputStream.write(buffer, 0, bytesRead);
+                 bytesRead = resourceStream.read(buffer);
+             }
+
+             byte[] eclData = byteArrayOutputStream.toByteArray();
+             String ecl = new String(eclData, "UTF-8");
+
+             WorkunitWrapper wu = new WorkunitWrapper();
+             wu.setECL(ecl);
+             wu.setJobname(eclScriptName);
+             wu.setCluster(thorclustername);
+
+             //wswuclient.createAndRunWUFromECLAndGetResults(wu);
+             WUPublishWorkunitResponseWrapper resp = wswuclient.publishWUFromEclWrapped(wu);
+             System.out.println("Finished publishing query" + resp.toString());
+             List<QueryResultWrapper> queries = wswuclient.listQueries(wu.getWuid(), wu.getJobname(), wu.getCluster(), null, null, null, null, null, null);
+             for (QueryResultWrapper query : queries)
+             {
+                 if (query.getName().equalsIgnoreCase(eclScriptName))
+                 {
+                     hasPublishedQuery = true;
+                     return;
+                 }
+             }
+        }
+        catch (Exception e)
+        {
+            System.out.println("Could not publish ECL query: " + e.getLocalizedMessage());
+        }
+
+        wseclConn = wsclient.getConnection();
+        wseclConn.setPort(wsECLPort);
+        wseclConn.setSocketTimeoutMilli(15000);
+        wseclConn.setWriteTimeoutMilli(15000);
+    }
+
+    @Test
+    public void testWsECLGetWSDL()
+    {
+        assumeTrue("WsECL connection not available", wseclConn != null);
+        assumeTrue("WsECL connection port appears invalid", wseclConn.getPort().equals(wsECLPort));
+        assumeTrue("Cannot test WsECL published query WSDL feature without published queries!", hasPublishedQuery);
+
+        try
+        {
+            String wsdlURI = "/WsEcl/definitions/query/" + thorclustername + "/" + eclScriptName + "/main/" + eclScriptName + ".wsdl";
+            
+            String wsdlResponse = wseclConn.sendGetRequest(wsdlURI);
```

Comment:
simple http get request based on the wsecl connection setup above, and the target URI constructed here.